core/src/main/java/dev/morphia/mapping/codec/MorphiaCodecProvider.java

```diff
@@ -48,11 +48,19 @@ public MorphiaCodecProvider(Datastore datastore) {
         this.datastore = datastore;
         this.mapper = datastore.getMapper();
 
+        CodecProvider codecProvider;
```

Comment:
so this is what I thought.  This change is unrelated to the lifecycle fix and should go in a different PR.  This change can't go in 2.3.1 as it introduces an API change.

Reply:
I understand, but I'll note that without this change I believe the project is broken and so this change should follow soon. Without something like this, custom codecs are ignored for the default codecs. Is it actually an API change to have registered custom codecs actually getting used? That's not so much an API change as a behavioral fix.